lib/notifications/display.dart

```diff
@@ -231,9 +233,21 @@ class NotificationDisplayManager {
   static Future<void> _onMessageFcmMessage(MessageFcmMessage data, Map<String, dynamic> dataJson) async {
     assert(debugLog('notif message content: ${data.content}'));
     final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
-    final groupKey = _groupKey(data);
+    final groupKey = _groupKey(data.realmUrl, data.userId);
     final conversationKey = _conversationKey(data, groupKey);
 
+    final globalStore = await ZulipBinding.instance.getGlobalStore();
+    final account = globalStore.accounts.firstWhereOrNull((account) =>
+      account.realmUrl.origin == data.realmUrl.origin && account.userId == data.userId);
+
+    // If the account is null, it means the user is logged out. This can happen
+    // if the unregisterToken request failed previously, leaving the account
+    // in a state where it's no longer valid in our system. In this scenario,
+    // we simply return and do nothing, as there's no active account to process.
+    if (account == null) {
+      return;
```

Comment:
I'd like this comment to focus more on the experience for the user.

How about this, adapted from the issue description on #1264:

```dart
    // Skip showing notifications for a logged-out account. It would be annoying
    // to the user if notifications keep showing up after they've logged out.
    // (Also alarming: it suggests the logout didn't fully work.)
```

Reply:
Looks good to me! But also, that doesn't explain the actual situation well, though

Reply:
Can you say more about how that differs from the actual situation?

Reply:
Surely, the comment focuses on the user's experience, but then hides why we had to check this for logged out accounts in the first place. It's because `unregisterToken` failed due to some reason.

Reply:
Thanks. Rereading the context, I think two things are both helpful to explain here: (a) why this is a situation that can arise (and isn't a sign of something buggy happening before this point, and (b) why just cheerfully doing nothing is an appropriate way to handle it.

So I think one reason Chris made the remark above is likely that this:
> In this scenario, we simply return and do nothing, as there's no active account to process.

can naturally make the reader wonder about question (b): do nothing? is that really what the user wants? we got a notification for them, after all.

I think it'd be quite reasonable to add back (relative to Chris's draft above) a mention of the unregisterToken request failing, though. That helps clarify the answer to question (a).